### PR TITLE
Initial check in.

### PR DESCRIPTION
**Pull request by evenflow58**

### CODE REVIEW

File: src/element/aureliaModularGrid.ts [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+        if(sortedField) {
+            switch(sortedField.direction) {
+                case SortDirection.Default:
+                    sortedField.direction = SortDirection.Asc;
+                    break;
+                case SortDirection.Asc:
+                    sortedField.direction = SortDirection.Desc;
+                    break;
+                case SortDirection.Desc:
+                    sortedField.direction = SortDirection.Default;
```
Comment (VladislavSudakov) [inline src/element/aureliaModularGrid.ts]: SortDirection.Desc ?

File: src/element/aureliaModularGrid.ts [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+     * @memberOf AureliaModularGrid
+     */
+    private sort(field: string): void {
+        if(!this.sortings) {
+            return;
+        }
+
+        let sortedField = this.sortings.find((s: Sort) => s.field === field);
+        if(sortedField) {
+            switch(sortedField.direction) {
```
Comment (VladislavSudakov) [inline src/element/aureliaModularGrid.ts]: I really dont understand what this code does, why need to make desc if direction is asc? why need to have that all switch case ?

File: src/element/aureliaModularGrid.ts [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+    }
+
+    /**
+     * Called by Aurelia when the page is being torn down.
+     * 
+     * @memberOf AureliaModularGrid
+     */
+    detached() {
+        this.observers.forEach((observer) => {
+            observer.dispose();
```
Comment (VladislavSudakov) [inline src/element/aureliaModularGrid.ts]: nice

File: src/models/column.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+     * Creates an instance of Column.
+     * 
+     * @param {string} field
+     * @param {string} friendlyName
+     * @param {number} order
+     * 
+     * @memberOf Column
+     */
+    constructor(field: string, friendlyName: string, order: number) {
+        this.field = field;
```
Comment (VladislavSudakov) [inline src/models/column.ts]: you can make public in ctor and save some space)

File: LICENSE [hunk 5]
```diff
@@ -0,0 +1,21 @@
+The MIT License (MIT)
+
+Copyright (c) 2016 Dwayne Charrington
```
Comment (gwely) [inline LICENSE]: you should update this

File: gulpfile.babel.js [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+const typescriptPattern = "./src/**/*.ts";
+const destination = "dist";
+
+const destinations = [
+    "dist/amd",
+    "dist/commonjs",
+    "dist/es2015",
+    "dist/system"
+];
+
```
Comment (gwely) [inline gulpfile.babel.js]: does this file need to be cleaned up too? are you using all of these tasks?

File: src/element/aureliaModularGrid.ts (hunk 1 above)
Comment (evenflow58) [inline src/element/aureliaModularGrid.ts]: Creates a loop of the enums. So it'll sort by asc on first click, desc and then default on successive clicks.

File: src/element/aureliaModularGrid.ts (hunk 2 above)
Comment (evenflow58) [inline src/element/aureliaModularGrid.ts]: See previous comment. This allows for cycling between sort types. I'll change the function name to make this more clear.

File: LICENSE (hunk 5 above)
Comment (evenflow58) [inline LICENSE]: Good catch. Thanks!

File: gulpfile.babel.js (hunk 6 above)
Comment (evenflow58) [inline gulpfile.babel.js]: All but the watch task.

**Pull request merged**